### PR TITLE
feat(refresh-token-rotation): Implement refresh token rotation

### PR DESCRIPTION
**Pull request by viacheslavkol**
### Purpose
Implement refresh token rotation
[FST-61](https://issues.folio.org/browse/FST-61)

### Approach
- Change login endpoint to `/login-with-expiry`
- Add refresh token endpoint
- Add tokens expiration checks
- Add tokens rotation

### TODOs and Open Questions
- [ ] Verify solution is valid after original PR in mod-login is merged

### Learning
[mod-login PR](https://github.com/folio-org/mod-login/pull/133)
[Rotation docs](https://wiki.folio.org/pages/viewpage.action?pageId=96414255)
Permission update not needed because `login` endpoints doesn't require one.

### CODE REVIEW

**Comment by viacheslavkol** [comment]
Do not merge until original PR merged + related dependencies` PRs are ready

**Comment by viacheslavkol** [comment]
Merge in scope of [MSEARCH-552](https://issues.folio.org/browse/MSEARCH-552)

**Review by hasanepam — APPROVED** [review]

**Comment by viacheslavkol** [comment]
Refresh token rotation functionality moved to folio-spring-support. This PR only extends OkapiExtension functionality

**Pull request merged**
